### ISSUE TITLE
Shrinking the Pixi lockfile by using a hash map

### ISSUE THREAD

**Issue by RaphaelMelanconAtBentley** [prefix-dev/pixi#2693]
### Problem description

I noticed that there is a lot of redundancy in the Pixi lockfile.
For instance, there is a unique entry for each package at the end of the lockfile, but then for each environment, the package's source is repeated in full.

Wouldn't it be efficient to use a hash map to assign a small id to each of those unique packages and use that small id in the environments section?
I think this would reduce the size considerably when a project has tons of environments sharing features and solve groups.
Not sure about the slowdowns involved with the extra steps parsing/creating that map.
This could also be done only if a certain threshold of redundancy is met.

Just my two cents.

**Comment by baszalmstra** [comment]
Yes that would help! But using ids would also make the lockfile less readable which is a big consideration too.

We have some other ideas though to further reduce it though! Especially around getting rid of the list of packages per environment per platform.

**Comment by olivier-lacroix** [comment]
I agreee readability is most important! Repeated entries will be compressed really well (incl by git) and are unlikely to matter that much?
